processing/src/main/java/org/apache/carbondata/processing/newflow/steps/input/InputProcessorStepImpl.java

```diff
@@ -0,0 +1,171 @@
+          output[k++] = fields[j];
+          break;
+        }
+      }
+    }
+    return output;
+  }
+
+  @Override public void initialize() throws CarbonDataLoadingException {
+    DataField[] output = getOutput();
+    genericParsers = new GenericParser[output.length];
+    for (int i = 0; i < genericParsers.length; i++) {
+      genericParsers[i] = CarbonParserFactory.createParser(output[i].getColumn(),
+          (String[]) configuration
+              .getDataLoadProperty(DataLoadProcessorConstants.COMPLEX_DELIMITERS));
+    }
+  }
+
+  private int getNumberOfCores() {
+    int numberOfCores;
+    try {
+      numberOfCores = Integer.parseInt(CarbonProperties.getInstance()
+          .getProperty(CarbonCommonConstants.NUM_CORES_LOADING,
+              CarbonCommonConstants.NUM_CORES_DEFAULT_VAL));
+    } catch (NumberFormatException exc) {
+      numberOfCores = Integer.parseInt(CarbonCommonConstants.NUM_CORES_DEFAULT_VAL);
+    }
+    return numberOfCores;
+  }
+
+  private int getBatchSize() {
+    int batchSize;
+    try {
+      batchSize = Integer.parseInt(configuration
+          .getDataLoadProperty(DataLoadProcessorConstants.DATA_LOAD_BATCH_SIZE,
+              DataLoadProcessorConstants.DATA_LOAD_BATCH_SIZE_DEFAULT).toString());
+    } catch (NumberFormatException exc) {
+      batchSize = Integer.parseInt(DataLoadProcessorConstants.DATA_LOAD_BATCH_SIZE_DEFAULT);
+    }
+    return batchSize;
+  }
+
+  @Override public Iterator<CarbonRowBatch>[] execute() {
+    int batchSize = getBatchSize();
+    List<Iterator<Object[]>>[] readerIterators = partitionInputReaderIterators();
+    Iterator<CarbonRowBatch>[] outIterators = new Iterator[readerIterators.length];
+    for (int i = 0; i < outIterators.length; i++) {
+      outIterators[i] = new InputProcessorIterator(readerIterators[i], genericParsers, batchSize);
+    }
+    return outIterators;
+  }
+
+  private List<Iterator<Object[]>>[] partitionInputReaderIterators() {
+    int numberOfCores = getNumberOfCores();
+    if (inputIterators.size() < numberOfCores) {
+      numberOfCores = inputIterators.size();
+    }
+    List<Iterator<Object[]>>[] iterators = new List[numberOfCores];
+    for (int i = 0; i < numberOfCores; i++) {
+      iterators[i] = new ArrayList<>();
+    }
+
+    for (int i = 0; i < inputIterators.size(); i++) {
+      iterators[i % numberOfCores].add(inputIterators.get(i));
+
+    }
+    return iterators;
+  }
+
+  @Override protected CarbonRow processRow(CarbonRow row) {
+    return null;
+  }
+
+  private static class InputProcessorIterator extends CarbonIterator<CarbonRowBatch> {
+
+    private List<Iterator<Object[]>> inputIterators;
+
+    private GenericParser[] genericParsers;
+
+    private Iterator<Object[]> currentIterator;
+
+    private int counter;
+
+    private int batchSize;
+
+    public InputProcessorIterator(List<Iterator<Object[]>> inputIterators,
+        GenericParser[] genericParsers, int batchSize) {
```

Comment:
It is consistent as generic parsers are  created from `DataField[]` returned by `getOutput`.